pyccel/parser/base.py

```diff
@@ -500,17 +512,118 @@ def insert_import(self, expr):
                         container[source] = []
                     container[source] += name
 
-    def print_namespace(self):
+    def dump(self, filename=None):
```

Comment:
It is not clear to me how the "filename" argument is used in this function. Can you explain please?

Reply:
if a filename is given you can then dump the parser in that file, if not you take the same name of the file that you are parsing